MicroLite/SqlUtility.cs

```diff
@@ -119,16 +120,17 @@ public static string RenumberParameters(string sql, int totalArgumentCount)
             var argsAdded = 0;
             var parameterPrefix = parameterNames[0].Substring(0, parameterNames[0].IndexOfAny(digits));
 
-            var predicateReWriter = new StringBuilder(sql);
+            var result = sql;
 
-            foreach (var parameterName in parameterNames.OrderByDescending(n => n))
+            foreach (var parameterName in parameterNames.OrderByDescending(n => n.Replace(parameterPrefix, string.Empty).PadLeft(9, '0')))
```

Comment:
This also results in a lot of extra string allocation and manipulation, a custom string comparer here would be more performant